compiler/transf.nim

```diff
@@ -671,9 +671,8 @@ proc transformFor(c: PTransf, n: PNode): PNode =
       idNodeTablePut(newC.mapping, formal, arg)
       # XXX BUG still not correct if the arg has a side effect!
     of paComplexOpenarray:
-      let typ = newType(tySequence, formal.owner)
-      addSonSkipIntLit(typ, formal.typ[0])
-      var temp = newTemp(c, typ, formal.info)
+      # arrays will deep copy here (pretty bad).
+      var temp = newTemp(c, arg.typ, formal.info)
```

Comment:
Temporaries of the type ``tyOpenArray`` are currently unsupported in the backends.

Reply:
I know. The temp will be a ``seq`` or an ``array``, not ``tyOpenArray``, after all it is inline. And if the iterator iterates an ``openarray`` that comes from an argument of a proc, then it will be covered by ``paDirectMapping``.

Reply:
The temporary will be of type seq or array here. After all it is inline and we know the the type before the conversation to openarray. Argument symbols that are truly openarray symbols should be coverd by ``paDirectMapping``. After all this assignment to a seq just generated incorrect code where an array was assigned to a seq variables. The C compiler was unhappy about it accepted it and bad things happened.